File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    pit=[]
    answer=0
    
    for i in moves:
        for j in range(len(board)):
            if board[j][i-1]!=0:
                pit.append(board[j][i-1])
                board[j][i-1]=0
                if len(pit)>1:
                    if pit[-1]==pit[-2]:
                        pit[-2:]=[]
                        answer+=2
                break
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
    private static int answer = 0;

    public int solution(int[][] board, int[] moves) {
        Stack<Integer> basket = new Stack<>();
        for (int i = 0; i < moves.length; i++) {
            grap(board, moves[i] - 1, basket);
        }
        return answer;
    }

    private void grap(int[][] board, int pos, Stack<Integer> basket) {
        for (int i = 0; i < board.length; i++) {
            if (board[i][pos] != 0) {
                placeDoll(board[i][pos], basket);
                board[i][pos] = 0;
                return;
            }
        }
    }

    private void placeDoll(int dollType, Stack<Integer> basket) {
        if (basket.empty() || basket.peek() != dollType) {
            basket.push(dollType);
        } else {
            answer += 2;
            basket.pop();
        }
    }
}